src/main/java/duke/command/DeadlineCommand.java

```diff
@@ -0,0 +1,100 @@
+package duke.command;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.exception.CalendarException;
+import duke.exception.DeadlineException;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+
+public class DeadlineCommand extends Command {
+
+    public DeadlineCommand(String task) {
+        super(task);
+    }
+
+    /**
+     * Executes any command corresponding to Deadline keyword.
+     * @param taskList List of tasks.
+     * @param ui       UI of the bot.
+     * @param storage  Storage managing the file in hard disk.
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) {
+        try {
+            return processDeadline(this.task, taskList, ui, storage);
+        } catch (DeadlineException dead) {
+            return dead.getMessage();
+        }
+    }
+
+    /**
+     * Processes all the deadline command to determine the correct output.
+     * @param theRest  Parsed string containing task details.
+     * @param taskList List containing all the task(s).
+     * @param ui       UI of the bot
+     * @param storage  Storage managing the file in hard disk.
+     * @throws DeadlineException If user's input is incomplete or in the wrong format.
+     */
+
+    public String processDeadline(
```

Comment:
Perhaps you could avoid deep nesting in this method body? I noticed the same issue in several other classes as well.